core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -232,6 +232,11 @@ class HadoopRDD[K, V](
         logWarning(s"${jobConf.get(FileInputFormat.INPUT_DIR)} doesn't exist and no" +
             s" partitions returned from this path.", e)
         Array.empty[Partition]
+      case e: IOException if e.getMessage.contains("Not a file") =>
```

Comment:
Nit: if e.getMessage.startsWith("Not a file:")

Reply:
> Nit: if e.getMessage.startsWith("Not a file:")

Done